src/librbd/io/ObjectRequest.cc

```diff
@@ -109,6 +110,23 @@ void ObjectRequest<I>::complete(int r)
     if (m_hide_enoent && r == -ENOENT) {
       r = 0;
     }
+    if (strcmp(get_op_type(), "cmpext") == 0 && r <= -MAX_ERRNO) {
+      ImageCtx *image_ctx = this->m_ictx;
+      uint64_t offset = -MAX_ERRNO - r;
+
+      __u32 object_size = image_ctx->layout.object_size;
+      __u32 su = image_ctx->layout.stripe_unit;
+      __u32 stripe_count = image_ctx->layout.stripe_count;
+      assert(object_size >= su);
+      uint64_t stripes_per_object = object_size / su;
+      uint64_t off_in_block = offset % su;
+      uint64_t stripepos = this->m_object_no % stripe_count;
+      uint64_t objectsetno = this->m_object_no / stripe_count;
+      uint64_t stripeno = offset / su + objectsetno * stripes_per_object;
+      uint64_t blockno = stripeno * stripe_count + stripepos;
+      uint64_t extent_off = blockno * su + off_in_block;
+      r = -MAX_ERRNO - extent_off;
```

Comment:
I think ssize_t is OK for librbd API. In 64-bit system, uint64_t is 8byte, and ssize_t is 8byte too. we only need  to change the int to ssize_t
If add an optional uint64_t pointer, we should assign the pointer in callback function, from objrequest to imagerequest, it is more complex